// telegram-format/dist/source/html.js
"use strict";
// https://core.telegram.org/bots/api#html-style
Object.defineProperty(exports, "__esModule", { value: true });
exports.html = void 0;
function escape(text) {
    return text
        .replace(/&/g, '&amp;')
        .replace(/</g, '&lt;')
        .replace(/>/g, '&gt;');
}
function bold(text) {
    return `<b>${text}</b>`;
}
function italic(text) {
    return `<i>${text}</i>`;
}
function strikethrough(text) {
    return `<s>${text}</s>`;
}
function underline(text) {
    return `<u>${text}</u>`;
}
function spoiler(text) {
    return `<span class="tg-spoiler">${text}</span>`;
}
function monospace(text) {
    return `<code>${escape(text)}</code>`;
}
function monospaceBlock(text, programmingLanguage) {
    if (programmingLanguage) {
        return `<pre><code class="language-${programmingLanguage}">${escape(text)}</code></pre>`;
    }
    return `<pre>${escape(text)}</pre>`;
}
function url(label, url) {
    return `<a href="${url}">${label}</a>`;
}
function userMention(label, userId) {
    return url(label, `tg://user?id=${userId}`);
}
exports.html = {
    parse_mode: 'HTML',
    escape,
    bold,
    italic,
    strikethrough,
    underline,
    spoiler,
    monospace,
    monospaceBlock,
    url,
    userMention,
};
