// telegram-format/dist/source/markdownv2.js
"use strict";
// https://core.telegram.org/bots/api#markdownv2-style
Object.defineProperty(exports, "__esModule", { value: true });
exports.markdownv2 = void 0;
function escapeInternal(text, escapeChars) {
    return text.replace(new RegExp(`[${escapeChars}\\\\]`, 'g'), '\\$&');
}
function escape(text) {
    return text.replace(/[_*[\]()~`>#+\-=|{}.!\\]/g, '\\$&');
}
function bold(text) {
    return `*${text}*`;
}
function italic(text) {
    return `_${text}_`;
}
function strikethrough(text) {
    return `~${text}~`;
}
function underline(text) {
    return `__${text}__`;
}
function spoiler(text) {
    return `||${text}||`;
}
function monospace(text) {
    return '`' + escapeInternal(text, '`') + '`';
}
function monospaceBlock(text, programmingLanguage) {
    let result = '';
    result += '```';
    if (programmingLanguage) {
        result += programmingLanguage;
    }
    result += '\n';
    result += escapeInternal(text, '`');
    result += '\n';
    result += '```';
    return result;
}
function url(label, url) {
    return `[${label}](${escapeInternal(url, ')')})`;
}
function userMention(label, userId) {
    return url(label, `tg://user?id=${userId}`);
}
exports.markdownv2 = {
    parse_mode: 'MarkdownV2',
    escape,
    bold,
    italic,
    strikethrough,
    underline,
    spoiler,
    monospace,
    monospaceBlock,
    url,
    userMention,
};
