// telegram-format/dist/source/markdown.js
"use strict";
// https://core.telegram.org/bots/api#markdown-style
Object.defineProperty(exports, "__esModule", { value: true });
exports.markdown = void 0;
function escape(text) {
    return text.replace(/[*_`[\]()]/g, '');
}
function bold(text) {
    return `*${text.replace(/\*/g, '')}*`;
}
function italic(text) {
    return `_${text.replace(/_/g, '')}_`;
}
function strikethrough(_text) {
    throw new Error('strikethrough is not supported by Markdown. Use MarkdownV2 instead.');
}
function underline(_text) {
    throw new Error('underline is not supported by Markdown. Use MarkdownV2 instead.');
}
function spoiler(_text) {
    throw new Error('spoiler is not supported by Markdown. Use MarkdownV2 instead.');
}
function monospace(text) {
    return '`' + text.replace(/`/g, '') + '`';
}
function monospaceBlock(text, programmingLanguage) {
    let result = '';
    result += '```';
    if (programmingLanguage) {
        result += programmingLanguage;
    }
    result += '\n';
    result += text.replace(/```/g, '');
    result += '\n';
    result += '```';
    return result;
}
function url(label, url) {
    return `[${label.replace(/]/, '')}](${url.replace(/\)/, '')})`;
}
function userMention(label, userId) {
    return url(label, `tg://user?id=${userId}`);
}
exports.markdown = {
    parse_mode: 'Markdown',
    escape,
    bold,
    italic,
    strikethrough,
    underline,
    spoiler,
    monospace,
    monospaceBlock,
    url,
    userMention,
};
